Objects/EventItem/Mag.gd

```diff
@@ -0,0 +1,45 @@
+extends "res://Objects/EventItem/EventItem.gd"
+
+onready var BONUS_TIME = 10
+
+var timer
+var default_damage_multiplier
+var default_damage
+
+
+func get_reward():
+	var stats = get_parent().find_node("Elf").stats
+	default_damage_multiplier = stats.damage_multiplier
+	stats.damage_multiplier *= 4.0
+	
+	var damage_stat = stats.get_stat("bows_knowledge")
+	damage_stat.set_default_value(damage_stat.default_value * stats.damage_multiplier)
+	default_damage = damage_stat.default_value * (stats.damage_multiplier/stats.damage_multiplier / 4.0)
+	damage_stat.calculate_changed_value()
+	
+	delayed_reset(BONUS_TIME)
+
+func delayed_reset(time):
+	timer = Timer.new()
+	timer.wait_time = time
+	timer.connect("timeout",self,"_on_timer_timeout")
+	timer.one_shot = true
```

Comment:
If you want to do 'one shot timer' I recommend to use yield function as shown here: https://docs.godotengine.org/en/3.1/getting_started/scripting/gdscript/gdscript_basics.html#coroutines-signals

Reply:
Yeee godot have yield as Unity ;)